kafka-connect/scripts/kafka_connect_run.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -e
+
+# Start bash if requested. Otherwise start Kafka Connect
+if [ "$1" = 'bash' ]; then
```

Comment:
Another noobie question, really, but why do we need this maybe-start-`bash` logic?

Reply:
For me this is useful for debugging. Without this option you always start the script and Kafka Connect and when it runs into error the image shutdowns. This option gives you a chance to start bash and from bash you can e.g.:
* Check the files which it downloaded
* Start Kafka connect manually
* Test the run script without rebuilding the image

The start command and entrypoint can be also overwritten from docker command line. But I never remember the syntax. This is easier. It is also common practice for many other images. 

*Anyway, I find it really useful, but if you think it should not be there, I can remove it.*

Reply:
@scholzj using the `docker run -it <image> /bin/bash` is well know by people using Docker (containers). What should be the way of using this "bash" parameter in this case ?

Reply:
@ppatierno Hmm, I never used `docker run -ti <image> /bin/bash` and never heard about it :-). I always used just `docker run -ti <image> bash`. But I guess there are different practices in place :-). The `bash` thing I know about comes from the guide to create official Docker Hub library images.

I think this could be done even better. We should use the `kafka_connect_run.sh` as `ENTRYPOINT` but also specify the `CMD`. And only when the original `CMD` is kept we proceed with our script. If anything else is specified we should run the specified command (see [here](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#entrypoint)). That way it will not be about `bash` or `/bin/bash` anymore.

Reply:
@scholzj you are right about just `bash` not `/bin/bash` but it works as well :-) Btw what should be the way of using the image with the script but running the bash. Can you give me an example ?

Reply:
@ppatierno For me it is just debugging thingy. I can start the image locally into bash and *debug* and *test* the kafka_connect_run.sh script (or the image per se - to check the files which were downloaded etc.). I can run it, I check the config file which it generated, I can modify it and immediately run it again without rebuilding the whole image. These things are a bit harder to do when you just start the image, it runs Kafka Connect and exits.

Reply:
BTW: OpenShift basically does something similar with the *debug the image/pod* feature. It is just less lazy and it overrides the entrypoint instead of relying on something in the entrypoint script it elf :-)